### DISCUSSION TITLE
Using api authentication using  API keys.

### DISCUSSION THREAD

**Discussion by vtkageyama** [mgmonteleone/python-atlasapi#153] [Ideas]
Hi, first congratulation by initiative, I found this API very good.

I would like know if you thinks in put authentication using API keys without using user/password ?

**Comment by mgmonteleone** [comment]
Hello!

Thanks for the question, and sorry for the long delay in answering.

Actually, the authentication is now done with API keys, even thought the names of the fields in the Atlas Object is `username` and `password`.

Sorry for the confusion, but the old "username" and "password" properties were from the very beginning of the project, and were never changed.

I may rename those fields in the upcoming 3.0 release to make this more clear!


Thanks for the support!

**Comment by mgmonteleone** [comment]
Change in pr #162 , and will be present in v3.0.1b2
